packages/grid/src/vaadin-grid-sort-mixin.js

```diff
@@ -126,22 +139,22 @@ export const SortMixin = (superClass) =>
     }
 
     /** @private */
-    __updateSorter(sorter) {
+    __updateSorter(sorter, shiftClick) {
       if (!sorter.direction && this._sorters.indexOf(sorter) === -1) {
         return;
       }
 
       sorter._order = null;
 
-      if (this.multiSort) {
+      if (this.multiSort || (this.multiSortOnShiftClick && shiftClick)) {
```

Comment:
Could add a test case for when a shift-click occurs but `multiSortOnShiftClick` is not enabled (should single-sort)

Reply:
Done.